src/streaming/controllers/ProtectionController.js

```diff
@@ -418,8 +421,62 @@ MediaPlayer.dependencies.ProtectionController = function () {
                 data:event.data,
                 error:null
             });
+        },
+        
+        // Resolve all init data, for Adobe Access the init may need to be downloaded from
+        // a given URL in the mpd.
+        getPsshForSupportedKeySystems = function(supportedKS) {
+
+            var self = this,
+                supportedKSInit = {};
+            supportedKSInit[MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE] = function(event) {
+                var ksIdx = 0,
+                    pending = false;
+                while( ksIdx < supportedKS.length) {
+                    if (event.data.requestData.schemeIdUri.toLowerCase() === supportedKS[ksIdx].ks.schemeIdURI) {
+                        if (!event.error) {
+                            supportedKS[ksIdx].initData = event.data.message;
+                        }
+                        else {
+                            self.log("DRM: Could not download PSSH data for keysystem : " + supportedKS[ksIdx].ks.schemeIdURI);
+                            supportedKS.splice(ksIdx);
+                            continue;
+                        }
+                        supportedKS[ksIdx].ks.unsubscribe(MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE, supportedKSInit);
+                    }
+                    else if (supportedKS[ksIdx].initData === "pending") {
+                        pending = true;
+                    }
+                    
+                    ksIdx++;
+                }
+                
+                if (!pending) {
+                    pendingPsshData = false;
+
+                    selectKeySystem.call(self, supportedKS, true);
+                }
+                
+            };
+            
+            if (supportedKS && supportedKS.length > 0) {
+                var ksIdx = 0;
+                
+                for(ksIdx = 0; ksIdx < supportedKS.length; ++ksIdx) {
+                    if (supportedKS[ksIdx].initData === "pending") {
+                        supportedKS[ksIdx].ks.subscribe(MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE, supportedKSInit);
+                        pendingPsshData = true;
+                    }
+                }
+            }
+            
+            if (!pendingPsshData) {
+                selectKeySystem.call(self, supportedKS, true);
+            }
+
         };
```

Comment:
@rashmimittal Since Primetime is the only DRM system we have encountered so far that allows for remote retrieval of PSSH boxes, I think we should limit this function to Primetime-specific code (i.e. `KeySystem.Primetime`).  We want to allow applications to build our library to include or exclude any DRM.  That is why the inclusion/selection of DRM systems is done in ProtectionExtensions.js, which is meant to be modified/replaced by implementers.  However, since this functionality makes `KeySystem.getInitData()` asynchronous, we might have to make all of the key system implementations of this function be asynchronous.  With our current Dijon event mechanism, I don't really like how that would look (adding `subscribe` and `notify` members to all).  If we could use W3C Promises, it would make the API much cleaner.  I might need to think about this one.